tests/nix_tests/ibc_utils.py

```diff
@@ -34,74 +50,91 @@
         "denom": "uatom",
     },
 }
+EVM_CHAINS = ["evmos_9000", "chainmain-1"]
 
 
 class IBCNetwork(NamedTuple):
     chains: Dict[str, Any]
     hermes: Hermes
 
 
-def prepare_network(
+def get_evmos_generator(
     tmp_path: Path,
     file: str,
-    other_chains_names: List[str],
-    custom_scenario=None,
+    is_rocksdb: bool = False,
+    custom_scenario: str | None = None,
 ):
-    file = f"configs/{file}.jsonnet"
-    if custom_scenario is not None:
-        # build the binary modified for a custom scenario
-        # e.g. allow to register WEVMOS token
-        # (removes a validation check in erc20 gov proposals)
-        cmd = [
-            "nix-build",
-            "--no-out-link",
-            str(Path(__file__).parent / f"configs/{custom_scenario}.nix"),
-        ]
-        print(*cmd)
-        modified_bin = (
-            Path(
-                subprocess.check_output(cmd, universal_newlines=True, text=True).strip()
-            )
-            / "bin/evmosd"
-        )
-        print(f"patched bin: {modified_bin}")
+    """
+    setup evmos with custom config
+    depending on the build
+    """
+    if is_rocksdb:
+        file = memiavl_config(tmp_path, file)
         gen = setup_custom_evmos(
             tmp_path,
-            26700,
+            26710,
             Path(__file__).parent / file,
-            post_init=update_evmos_bin(modified_bin),
-            chain_binary=modified_bin,
+            chain_binary="evmosd-rocksdb",
+            post_init=create_snapshots_dir,
         )
     else:
-        gen = setup_custom_evmos(tmp_path, 26700, Path(__file__).parent / file)
+        file = f"configs/{file}.jsonnet"
+        if custom_scenario is not None:
```

Comment:
nit: this would also catch an empty string I think
```suggestion
        if custom_scenario:
```